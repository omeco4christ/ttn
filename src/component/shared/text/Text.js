import styled from '@emotion/native';
import PropType from 'prop-types';
import {COLOR_TEXT_PRIMARY} from '../../../constants/styles';

const fontWeight = {
  400: 'AvenirLTStd-Roman',
  700: 'AvenirLTStd-Black',
};

export const TextExLg = styled.Text`
  font-size: 34px;
  font-family: ${props =>
    props.weight ? fontWeight[props.weight] : fontWeight['400']};
  color: ${props => props.color || COLOR_TEXT_PRIMARY};
`;

export const TextLg = styled.Text`
  font-size: 28px;
  font-family: ${props =>
    props.weight ? fontWeight[props.weight] : fontWeight['400']};
  color: ${props => props.color || COLOR_TEXT_PRIMARY};
`;

export const TextMd = styled.Text`
  font-size: 22px;
  font-family: ${props =>
    props.weight ? fontWeight[props.weight] : fontWeight['400']};
  color: ${props => props.color || COLOR_TEXT_PRIMARY};
`;

export const TextNorm = styled.Text`
  font-size: 17px;
  font-family: ${props =>
    props.weight ? fontWeight[props.weight] : fontWeight['400']};
  color: ${props => props.color || COLOR_TEXT_PRIMARY};
`;

export const TextSm = styled.Text`
  font-size: 15px;
  font-family: ${props =>
    props.weight ? fontWeight[props.weight] : fontWeight['400']};
  color: ${props => props.color || COLOR_TEXT_PRIMARY};
  line-height: 24px;
  text-align: ${props => props.textAlign || 'left'};
`;

export const TextXs = styled.Text`
  font-size: 13px;
  font-family: ${props =>
    props.weight ? fontWeight[props.weight] : fontWeight['400']};
  color: ${props => props.color || COLOR_TEXT_PRIMARY};
  line-height: 24px;
`;

TextExLg.propTypes = {
  color: PropType.string,
  weight: PropType.string,
};
TextLg.propTypes = {
  color: PropType.string,
  weight: PropType.string,
};

TextMd.propTypes = {
  color: PropType.string,
  weight: PropType.string,
};

TextSm.propTypes = {
  color: PropType.string,
  weight: PropType.string,
};
TextXs.propTypes = {
  color: PropType.string,
  weight: PropType.string,
};
