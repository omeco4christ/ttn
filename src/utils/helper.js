import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';

dayjs.extend(relativeTime);

export const logger = console.log;

export const htmlToText = html => html.replace(/<(.*?)>/g, '');

export const timeFromNow = date => dayjs(date).fromNow();

export const displayDate = date => dayjs(date).format('MMM YYYY');
