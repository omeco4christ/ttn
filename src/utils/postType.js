class PostType {
  constructor() {
    this.types = {
      question: 'Ask Question',
      share_link: 'Share a link',
      forum_post: 'Post to Forum',
    };
  }

  getType(type) {
    return this.types[type];
  }

  keysArray() {
    return Object.keys(this.types);
  }

  questionType() {
    return {key: this.keysArray()[0], value: this.types[this.keysArray()[0]]};
  }

  shareLinkType() {
    return {key: this.keysArray()[1], value: this.types[this.keysArray()[1]]};
  }

  forumType() {
    return {key: this.keysArray()[2], value: this.types[this.keysArray()[2]]};
  }
}

const postType = new PostType();

export default postType;
