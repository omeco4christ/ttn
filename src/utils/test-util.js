import React from 'react';
import renderer from 'react-test-renderer';

const AllTheProviders = ({children}) => {
  return <React.Fragment>{children}</React.Fragment>;
};

const customRender = (ui, options) =>
  renderer(ui, {wrapper: AllTheProviders, ...options});

export {customRender as renderer};
