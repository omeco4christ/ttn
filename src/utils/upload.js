import RNFetchBlob from 'rn-fetch-blob';
import {launchImageLibrary} from 'react-native-image-picker';

export const uploadImageFromStorage = () => {
  return new Promise((resolve, reject) => {
    launchImageLibrary(
      {includeBase64: true, maxHeight: 850, maxWidth: 720, mediaType: 'photo'},
      async response => {
        const image = response.assets[0];
        resolve(image);
      },
    );
  });
};

export const uploadImageToServer = image => {
  return new Promise((resolve, reject) => {
    RNFetchBlob.fetch(
      'POST',
      'https://dev-api.tellit.io/api/v1/upload',
      {
        'Content-Type': 'multipart/form-data',
      },
      [
        {
          name: 'file',
          filename: image.fileName,
          type: image.type,
          data: RNFetchBlob.wrap(image.uri),
        },
      ],
    )
      .then(res => {
        let parsedData = JSON.parse(res.data);
        console.log(parsedData.data);
        if (parsedData) {
          resolve(parsedData.data.url);
        }
      })
      .catch(err => reject(err));
  });
};

export const uploadImageFromStorageToServer = async () => {
  const imageFromStorage = await uploadImageFromStorage();
  return await uploadImageToServer(imageFromStorage);
};
