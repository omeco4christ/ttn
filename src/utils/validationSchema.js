import * as Yup from 'yup';

export const loginSchema = Yup.object({
  username: Yup.string().required('Username or email is required'),
  password: Yup.string().required('Password is required'),
});

export const signupSchema = Yup.object({
  first_name: Yup.string().required('First name is required'),
  last_name: Yup.string().required('Last name is required'),
  username: Yup.string().required('Username is required'),
  phone_number: Yup.number().required('Phone number is required'),
  email: Yup.string().email('Invalid email').required('Email is required'),
  password: Yup.string()
    .min(6, 'Password must be at least 6 character')
    .required('Password is required.'),
  confirm_password: Yup.string().oneOf(
    [Yup.ref('password'), null],
    'Passwords not match',
  ),
});

export const forgetPasswordSchema = Yup.object({
  email: Yup.string().email('Invalid email').required('Email is required'),
});

export const resetPasswordSchema = Yup.object({
  password: Yup.string()
    .min(6, 'Password must be at least 6 character')
    .required('Password is required.'),
  confirm_password: Yup.string().oneOf(
    [Yup.ref('password'), null],
    'Passwords not match',
  ),
});

export const resetPasswordTokenSchema = Yup.object({
  token: Yup.number()
    .required('Token is required')
    .max(9999, 'Token must be 4 length'),
});

export const changePasswordSchema = Yup.object({
  old_password: Yup.string().required('Old password is required.'),
  new_password: Yup.string()
    .min(6, 'Password must be at least 6 character')
    .required('New password is required.'),
  confirm_password: Yup.string().oneOf(
    [Yup.ref('new_password'), null],
    'Passwords not match',
  ),
});

export const forumTopicSchema = Yup.object({
  title: Yup.string().required('Title is required'),
  description: Yup.string().required('Description is required'),
});

export const forumTopicCommentSchema = Yup.object({
  comment: Yup.string().required('Comment is required'),
});
