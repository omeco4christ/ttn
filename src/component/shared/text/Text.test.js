import React from 'react';
import {render} from '../../../utils/test-util';
import {expect} from '@jest/globals';
import {TextExLg, TextLg, TextMd, TextSm} from './Text';
import renderer from 'react-test-renderer';

describe('Text Components', () => {
  it('Extra Large Text renders correctly', () => {
    const wrapper = renderer.create(<TextExLg />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('Large Text renders correctly', () => {
    const wraper = renderer.create(<TextLg />).toJSON();
    expect(wraper).toMatchSnapshot();
  });
  it('Medium Text renders correctly', () => {
    const wrapper = renderer.create(<TextMd />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('Small Text renders correctly', () => {
    const wrapper = renderer.create(<TextSm />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
