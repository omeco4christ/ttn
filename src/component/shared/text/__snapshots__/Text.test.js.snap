// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Components Extra Large Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
        "fontSize": 34,
        "fontWeight": "900",
      },
      undefined,
    ]
  }
/>
`;

exports[`Text Components Large Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
        "fontSize": 28,
        "fontWeight": "900",
      },
      undefined,
    ]
  }
/>
`;

exports[`Text Components Medium Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
        "fontSize": 22,
        "fontWeight": "800",
      },
      undefined,
    ]
  }
/>
`;

exports[`Text Components Small Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
        "fontSize": 15,
        "fontWeight": "400",
      },
      undefined,
    ]
  }
/>
`;
